packages/ui/src/form/DynamicFormField/DynamicInput.tsx

```diff
@@ -0,0 +1,172 @@
+import { FormFieldItem } from '@bigcommerce/checkout-sdk';
+import { isDate, noop } from 'lodash';
+import React, { FunctionComponent, memo, useCallback } from 'react';
+import ReactDatePicker from 'react-datepicker';
+
+import { useLocale } from '@bigcommerce/checkout/locale';
+
+import CheckboxInput from '../CheckboxInput/CheckboxInput';
+import { InputProps } from '../Input/Input';
+import RadioInput from '../RadioInput/RadioInput';
+import TextArea from '../TextArea/TextArea';
+import TextInput from '../TextInput/TextInput';
+
+import DynamicFormFieldType from './DynamicFormFieldType';
+
+export interface DynamicInputProps extends InputProps {
+    id: string;
+    additionalClassName?: string;
+    value?: string | string[];
+    rows?: number;
+    fieldType?: DynamicFormFieldType;
+    options?: FormFieldItem[];
+}
+
+const DynamicInput: FunctionComponent<DynamicInputProps> = ({
+    fieldType,
+    id,
+    name,
+    onChange = noop,
+    options,
+    placeholder,
+    value,
+    ...rest
+}) => {
+    const { date } = useLocale();
+    const { inputFormat } = date || { inputFormat: '' };
+    const handleDateChange = useCallback(
+        (dateValue: string, event) =>
+            onChange({
+                ...event,
+                target: {
+                    name,
+                    value: dateValue,
+                },
+            }),
+        [onChange, name],
+    );
+
+    switch (fieldType) {
+        case DynamicFormFieldType.DROPDOWM:
```

Comment:
Same here. Should be a lot of `useFloatingLabel` props in this file.

Reply:
Possibly, I only plan to use these for payment forms as of now. And since they don't have floating labels then should be ok.